packages/rich-text/src/plugins/Table/withInsertFragmentTable.ts

```diff
@@ -0,0 +1,166 @@
+              });
+            }
+          });
+
+          select(editor, {
+            anchor: getStartPoint(editor, cellEntries[0][1]),
+            focus: getEndPoint(editor, cellEntries.at(-1)![1]),
+          });
+          return;
+        }
+      }
+    }
+
+    if (insertedTable) {
+      const tableEntry = getTableAbove(editor, {
+        at: editor.selection?.anchor,
+      });
+
+      // inserting inside table
+      if (tableEntry) {
+        const [cellEntry] = getTableGridAbove(editor, {
+          at: editor.selection?.anchor,
+          format: 'cell',
+        });
+
+        if (cellEntry) {
+          withoutNormalizing(editor, () => {
+            const [, startCellPath] = cellEntry;
+            const cellPath = [...startCellPath];
+
+            const startColIndex = cellPath.at(-1)!;
+            let lastCellPath: Path | null = null;
+
+            let initRow = true;
+            const insertedRows = (insertedTable as TElement).children as TElement[];
+            insertedRows.forEach((row) => {
+              cellPath[cellPath.length - 1] = startColIndex;
+
+              // last inserted row
+              if (!initRow) {
+                const fromRow = cellPath.slice(0, -1);
+                cellPath[cellPath.length - 2] += 1;
+
+                if (!hasNode(editor, cellPath)) {
+                  if (disableExpandOnInsert) {
+                    return;
+                  } else {
+                    insertRow?.(editor, {
+                      fromRow,
+                    });
+                  }
+                }
+              }
+              initRow = false;
+
+              const insertedCells = row.children as TElement[];
+              let initCell = true;
+
+              insertedCells.forEach((cell) => {
+                if (!initCell) {
+                  const fromCell = [...cellPath];
+                  cellPath[cellPath.length - 1] += 1;
+
+                  if (!hasNode(editor, cellPath)) {
+                    if (disableExpandOnInsert) {
+                      return;
+                    } else {
+                      insertColumn?.(editor, {
+                        fromCell,
+                      });
+                    }
+                  }
+                }
+                initCell = false;
+
+                replaceNodeChildren(editor, {
+                  at: cellPath,
+                  nodes: cloneDeep(cell.children as any),
+                });
+
+                lastCellPath = [...cellPath];
+              });
+            });
+
+            if (lastCellPath) {
+              select(editor, {
+                anchor: getStartPoint(editor, startCellPath),
+                focus: getEndPoint(editor, lastCellPath),
+              });
+            }
+          });
+
+          return;
+        }
+      } else if (fragment.length === 1 && fragment[0].type === ELEMENT_TABLE) {
+        // NOTE adjusting this from insertNode to insertFragment to removing preceding empty paragraph
```

Comment:
I changed this instead of inserting as node in the original implementation inserting fragment